messaginginapp/src/test/java/io/customer/messaginginapp/InAppMessagingStoreTest.kt

```diff
@@ -0,0 +1,372 @@
+package io.customer.messaginginapp
+
+import io.customer.commontest.config.TestConfig
+import io.customer.commontest.config.testConfigurationDefault
+import io.customer.commontest.core.TestConstants
+import io.customer.commontest.extensions.attachToSDKComponent
+import io.customer.commontest.extensions.random
+import io.customer.messaginginapp.di.inAppMessagingManager
+import io.customer.messaginginapp.gist.GistEnvironment
+import io.customer.messaginginapp.gist.data.model.Message
+import io.customer.messaginginapp.state.InAppMessagingAction
+import io.customer.messaginginapp.state.InAppMessagingManager
+import io.customer.messaginginapp.state.MessageState
+import io.customer.messaginginapp.testutils.core.IntegrationTest
+import io.customer.messaginginapp.type.InAppEventListener
+import io.customer.messaginginapp.type.InAppMessage
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.data.model.Region
+import io.mockk.Called
+import io.mockk.confirmVerified
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.amshove.kluent.shouldBe
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeInstanceOf
+import org.amshove.kluent.shouldContain
+import org.amshove.kluent.shouldContainAll
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+class InAppMessagingStoreTest : IntegrationTest() {
+
+    private var inAppEventListener = mockk<InAppEventListener>(relaxed = true)
+
+    private val module = ModuleMessagingInApp(
+        config = MessagingInAppModuleConfig.Builder(
+            siteId = TestConstants.Keys.SITE_ID,
+            region = Region.US
+        ).setEventListener(inAppEventListener).build()
+    ).attachToSDKComponent()
+
+    private lateinit var manager: InAppMessagingManager
+
+    override fun setup(testConfig: TestConfig) {
+        super.setup(
+            testConfigurationDefault {
+                diGraph {
+                    sdk {
+                        overrideDependency<ModuleMessagingInApp>(module)
+                    }
+                }
+            }
+        )
```

Comment:
I think `attachToSDKComponent` does the same and this can be simplified to

```suggestion
        super.setup(testConfig)
```

Reply:
hmm makes sense, i would rather move the module here as well, otherwise doesn't make at top